drivers/include/rn2xx3.h

```diff
@@ -0,0 +1,699 @@
+/*
+ * Copyright (C) 2017 Inria
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    drivers_rn2xx3 RN2483/RN2903 LoRa module driver
+ * @ingroup     drivers_netdev
+ * @brief       High-level driver for the RN2483/RN2903 LoRa modules
+ * @{
+ *
+ * @file
+ * @brief       High-level driver for the RN2483/RN2903 LoRa modules
+ *
+ * @author      Alexandre Abadie <alexandre.abadie@inria.fr>
+ */
+
+#ifndef RN2XX3_H
+#define RN2XX3_H
+
+#include <stdint.h>
+
+#include "xtimer.h"
+#include "periph/uart.h"
+#include "periph/gpio.h"
+#include "net/netdev.h"
+
+#include "net/loramac.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   Maximum length of an exchanged messages with commands
+ */
+#define RN2XX3_MAX_BUF                  (40U)
+
+/**
+ * @brief   Maximum length of an RX message
+ *
+ *          LoRaMAC payload can be up to 250 bytes.
+ */
+#define RN2XX3_RX_MAX_BUF               (250U)
+
+/**
+ * @brief   Maximum delay in second to receive a reply from server.
+ */
+#define RN2XX3_REPLY_DELAY_TIMEOUT      (60U)
+
+/**
+ * @brief   Minimum sleep duration (in ms)
+ */
+#define RN2XX3_SLEEP_MIN                (100U)
+
+/**
+ * @brief   Default sleep duration (in ms)
+ */
+#ifndef RN2XX3_DEFAULT_SLEEP
+#define RN2XX3_DEFAULT_SLEEP            (5000U)
+#endif
+
+#if defined(MODULE_RN2903)
+#define RN2XX3_FREQ_BAND                (915U)
+#endif
+
+#if defined(MODULE_RN2483)
+#define RN2XX3_FREQ_BAND                (868U)
+#endif
+
+/**
+ * @brief   Command responses and server replies status codes
+ */
+enum {
```

Comment:
This looks a bit weird in many ways:
* mixing positiv and negative values in random fashion -> should be ordered
* also mixing error and (non-error) status/reply codes seems unwise

why note have an enum for the error codes, i.e., `RN2XX3_ERR_*`, and another for status reply cods, i.e. `RN2XX3_REPLY_*`?

Reply:
Because the same function can return either status codes, error codes and reply codes. This depends on the module behaviour.
Maybe I can just use one enum with only positive values?
I remember I couldn't find a clean solution (at least for me) for correctly handling the different error/status/reply codes.

Reply:
@smlng, I finally kept the single enum with all status codes inside but removed the mix of negative and positive values. I hope it will be fine.

Reply:
looks better now, but you still mix error codes, replies, and others - I would recommend to order/group them, i.e.,
```
enum {
    RN2XX3_OK,
    RN2XX3_DATA,
    RN2XX3_TIMEOUT,
    RN2XX3_ERR_MAC_INIT,
    /* all other err codes ... */
    RN2XX3_REPLY_TX_MAC_OK,
    /* all other reply codes ... */
}
```